vllm/worker/habana_model_runner.py

```diff
@@ -1087,9 +1087,10 @@ def prepare_input_tensors(
         batch_size_padded = find_bucket(real_batch_size, bucket_cfg)
         batch_size_padding = batch_size_padded - real_batch_size
         seq_group_metadata_list = seq_group_metadata_list.copy()
-        seq_group_metadata_list.extend(
-            self.create_dummy_seq_group_metadata(0, 0, is_prompt)
-            for _ in range(batch_size_padding))
+        dummy_seq_group_metadata = self.create_dummy_seq_group_metadata(
```

Comment:
let's add an 'if' so that we do it only when we know we need padding. Like:
```python
if batch_size_padding > 0:
    dummy_seq_group_metadata = self.create_dummy_seq_group_metadata(...)
    seq_group_metadata_list.extend(...)
```